### PR TITLE
Displacement Kalman Filter

### PR DESCRIPTION
**Pull request by rbhog**
Kept example kf as a stub to just show how to make a new kf. Moved displacement kf stuff into dedicated files and update in systems.cpp.

### CODE REVIEW

**Review by aidancostello — CHANGES_REQUESTED** [review]

File: MIDAS/src/systems.cpp [hunk 1]
```diff
@@ -158,9 +162,29 @@ DECLARE_THREAD(buzzer, RocketSystems* arg) {
  */
 DECLARE_THREAD(kalman, RocketSystems* arg) {
     example_kf.initialize();
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:164]: does this need to change to displacement kf init?

Comment (rbhog) [reply MIDAS/src/systems.cpp]: good catch, switched it

File: MIDAS/src/systems.cpp [hunk 2]
```diff
@@ -158,9 +162,29 @@ DECLARE_THREAD(buzzer, RocketSystems* arg) {
...
+        Acceleration current_accelerations = {
+            .ax = current_accelerometer.ax,
+            .ay = current_accelerometer.ay,
+            .az = current_accelerometer.az
+        };
+        // HighGData current_accelerometer = arg->rocket_state.high_g.getRecent();
+        // Orientation current_orientation = arg->rocket_state.orientation.getRecent();
+        // example_kf.kfTickFunction(current_state.curr_state, current_barometer_buf, current_accelerometer, current_orientation ,xTaskGetTickCount() - last);
+        float dt = (xTaskGetTickCount() - last);
+        displacement_kf.kfTick(dt, 13.0, current_barom_buf, current_accelerations);
```
[the remark this answers is not in the document]
Comment (rbhog) [reply MIDAS/src/systems.cpp:177]: added kalman filter data to rocket data

File: MIDAS/platformio.ini [hunk 3]
```diff
@@ -6,7 +6,7 @@ extra_scripts =
     pre:extra_script.py
 build_flags =
     -DARDUINO_USB_CDC_ON_BOOT=1
-    -std=gnu++17
+    -std=gnu++2a
```
Comment (redindelible) [inline MIDAS/platformio.ini:9]: Do newer versions of clang and gcc still support the gnu++2a thing or do you have to say gnu++20?

Comment (rbhog) [reply MIDAS/platformio.ini:9]: u have to say gnu++2a, gnu++20 didn't work

**Review by aidancostello — APPROVED** [review]

**Comment by aidancostello** [comment]
I agree with magilan, since you don't actually use any of the fsm stuff for this pr we can just delete

**Comment by rbhog** [comment]
removed all fsm stuff

**Review by redindelible — APPROVED** [review]

**Pull request merged**